pil-analyzer/src/evaluator.rs

```diff
@@ -440,24 +450,43 @@ impl<'a> Definitions<'a> {
                 .into()
             }
         } else {
-            match value {
-                Some(FunctionValueDefinition::Expression(TypedExpression {
-                    e: value,
-                    type_scheme,
-                })) => {
-                    let type_args = type_arg_mapping(type_scheme, type_args);
-                    evaluate_generic(value, &type_args, symbols)?
+            let impl_ = match type_args {
+                Some(type_args) => solved_impls
+                    .get(&name)
+                    .and_then(|inner_map| inner_map.get(type_args)),
```

Comment:
I think you need to first call `type_arg_mapping(type_scheme, type_args)` and then use the result here.

Reply:
Please also add a test to check that, i.e. call a trait function from a generic function and then at the very outside, call that generic function with two types.